src/components/docs/Code/index.js

```diff
@@ -0,0 +1,121 @@
+import React, {useEffect, useRef, useState} from "react";
```

Comment:
You might want to check out https://github.com/facebook/docusaurus/blob/master/packages/docusaurus-theme-classic/src/theme/CodeBlock/index.js
as another example of a codeblock.

This one ended up being quite different, but the basic logic in the render (line 86-104) is the same